File: Cpp/Check_If_Digits_Are_Equal_in_String_After_Operations_I.cpp
class Solution {
public:
    bool hasSameDigits(string s) {
        vector<int> digits;
        for (char c : s) {
            digits.push_back(c - '0');
        }
        int len = digits.size();
        while (len > 2) {
            for (int i = 0; i < len - 1; i++) {
                digits[i] = (digits[i] + digits[i + 1]) % 10;
            }
            len--;
        }
        return digits[0] == digits[1];
    }
};

File: Rust/Check_If_Digits_Are_Equal_in_String_After_Operations_I.rs
impl Solution {
    pub fn has_same_digits(s: String) -> bool {
        let mut digits: Vec<u32> = s.chars().map(|c|c.to_digit(10).unwrap()).collect();
        let mut len = digits.len();
        while len > 2{
            for i in 0..len - 1{
                digits[i] = (digits[i] + digits[i+1]) %10;
            }
            len -= 1;
        }
        digits[0] == digits[1]
    }
}